homeassistant/components/monzo/sensor.py

```diff
@@ -0,0 +1,124 @@
+
+    value_fn: Callable[[dict[str, Any]], Any]
+
+
+ACC_SENSORS = (
+    MonzoSensorEntityDescription(
+        key="balance",
+        translation_key="balance",
+        value_fn=lambda data: data["balance"]["balance"] / 100,
+        device_class=SensorDeviceClass.MONETARY,
+        native_unit_of_measurement="GBP",
+        suggested_display_precision=2,
+    ),
+    MonzoSensorEntityDescription(
+        key="total_balance",
+        translation_key="total_balance",
+        value_fn=lambda data: data["balance"]["total_balance"] / 100,
+        device_class=SensorDeviceClass.MONETARY,
+        native_unit_of_measurement="GBP",
+        suggested_display_precision=2,
+    ),
+)
+
+POT_SENSORS = (
+    MonzoSensorEntityDescription(
+        key="pot_balance",
+        translation_key="pot_balance",
+        value_fn=lambda data: data["balance"] / 100,
+        device_class=SensorDeviceClass.MONETARY,
+        native_unit_of_measurement="GBP",
+        suggested_display_precision=2,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Defer sensor setup to the shared sensor module."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][CONF_COORDINATOR]
+
+    accounts = [
+        MonzoSensor(
+            coordinator,
+            entity_description,
+            index,
+            account["name"],
+            lambda x: x["accounts"],
+        )
+        for entity_description in ACC_SENSORS
+        for index, account in enumerate(
+            hass.data[DOMAIN][config_entry.entry_id][ACCOUNTS]
+        )
+    ]
+
+    pots = [
+        MonzoSensor(
+            coordinator, entity_description, index, MODEL_POT, lambda x: x[POTS]
+        )
+        for entity_description in POT_SENSORS
+        for index, _pot in enumerate(hass.data[DOMAIN][config_entry.entry_id][POTS])
+    ]
+
+    async_add_entities(accounts + pots)
+
+
+class MonzoSensor(MonzoBaseEntity, SensorEntity):
+    """Represents a Monzo sensor."""
+
+    entity_description: MonzoSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        entity_description: MonzoSensorEntityDescription,
+        index: int,
+        device_model: str,
+        data_accessor: Callable[
+            [dict[str, list[dict[str, Any]]]], list[dict[str, Any]]
+        ],
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator, index, device_model, data_accessor)
+        self.entity_description = entity_description
+        self._attr_unique_id = f"{self.data['id']}_{entity_description.key}"
+
+    @property
+    def native_value(self) -> StateType:
+        """Return the state."""
+
+        try:
+            state = self.entity_description.value_fn(self.data)
+        except (KeyError, ValueError):
+            return None
+
+        return cast(StateType, state)
```

Comment:
Instead of this, could we type the value functions as returning `StateType` or `float`?